Location(ClassInfo)#toVariableNameString throws error when missing world

### Skript/Server Version

```markdown
> sk info
[07:53:16 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[07:53:16 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[07:53:16 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[07:53:16 INFO]: [Skript] Server Version: 26.1.2-DEV-20ab6cb (MC: 26.1.2)
[07:53:16 INFO]: [Skript] Skript Version: 2.15.2 (ketchup)
[07:53:16 INFO]: [Skript] Installed Skript Addons: 
[07:53:16 INFO]: [Skript]  - skNoise v1.0
[07:53:16 INFO]: [Skript]  - skript-reflect v2.6.3 (https://github.com/SkriptLang/skript-reflect)
[07:53:16 INFO]: [Skript]  - SkBee v3.24.0 (https://github.com/ShaneBeee/SkBee)
[07:53:16 INFO]: [Skript]  - SkBriggy v1.7.0 (https://github.com/ShaneBeee/SkBriggy)
[07:53:16 INFO]: [Skript]  - SkNMS v1.7.0-26.1.1 (https://github.com/ShaneBeee/SkNMS)
[07:53:16 INFO]: [Skript] Installed dependencies: 
[07:53:16 INFO]: [Skript]  - Vault v1.7.3-b131
```

### Bug Description

When a Location variable is missing a world and you use it as a var index, it'll throw a console error

### Expected Behavior

No error

### Steps to Reproduce

my whole test code:
```AppleScript
on right click on respawn anchor:
	set {-boom::%location of event-block%} to player

on block explode:
	wait 2 seconds
	# safe guard to prevent extra vars
	delete {-boom::%location of event-block%} 

on death of player:
	set {_loc} to damage location of damage source
	if {_loc} is set:
		set world of {_loc} to world of victim
		set {_boom} to {-boom::%{_loc}%} # <--- this is where the magic happens
		if {_boom} is set:
			send "%{_boom}% caused %victim% to die with explosion" to console

```

### Errors or Screenshots

```
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! [Skript] Severe Error:
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
[07:37:48 ERROR]: #!#! You are using some plugins that alter how Skript works (addons).
[07:37:48 ERROR]: #!#! Full list of addons:
[07:37:48 ERROR]: #!#! SkBriggy v1.7.0 (https://github.com/ShaneBeee/SkBriggy)
[07:37:48 ERROR]: #!#! skNoise v1.0
[07:37:48 ERROR]: #!#! SkBee v3.24.0 (https://github.com/ShaneBeee/SkBee)
[07:37:48 ERROR]: #!#! skript-reflect v2.6.3 (https://github.com/SkriptLang/skript-reflect)
[07:37:48 ERROR]: #!#! SkNMS v1.7.0-26.1.1 (https://github.com/ShaneBeee/SkNMS)
[07:37:48 ERROR]: #!#! CorePlugin v1.0.0 (https://github.com/ShaneBeee/CorePlugin)
[07:37:48 ERROR]: #!#! StressTestBots v1.1.0-26.1.2
[07:37:48 ERROR]: #!#! We could not identify related addons, it might also be a Skript issue.
[07:37:48 ERROR]: #!#! Try temporarily removing the listed plugins one by one to identify the cause.
[07:37:48 ERROR]: #!#! If removing a plugin resolves the issue, please report the problem to the plugin developer.
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! Stack trace:
[07:37:48 ERROR]: #!#! Caused by: java.lang.NullPointerException: Cannot invoke "org.bukkit.World.getName()" because the return value of "org.bukkit.Location.getWorld()" is null
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//org.skriptlang.skript.bukkit.types.LocationClassInfo$LocationParser.toVariableNameString(LocationClassInfo.java:135)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//org.skriptlang.skript.bukkit.types.LocationClassInfo$LocationParser.toVariableNameString(LocationClassInfo.java:120)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.classes.Parser.toString(Parser.java:68)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.registrations.Classes.toString(Classes.java:675)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.registrations.Classes.toString(Classes.java:702)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.registrations.Classes.toString(Classes.java:695)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.VariableString.toString(VariableString.java:377)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Variable.getRaw(Variable.java:349)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Variable.get(Variable.java:375)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Variable.getConverted(Variable.java:433)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Variable.getAll(Variable.java:750)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Variable.getArray(Variable.java:742)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.effects.EffChange.execute(EffChange.java:306)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Effect.run(Effect.java:42)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:40)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:67)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:33)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.SkriptEventHandler.lambda$execute$0(SkriptEventHandler.java:169)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.SkriptEventHandler.lambda$execute$1(SkriptEventHandler.java:180)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.util.Task.callSync(Task.java:212)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.util.Task.callSync(Task.java:196)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.SkriptEventHandler.execute(SkriptEventHandler.java:178)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:111)
[07:37:48 ERROR]: #!#!     at Skript-2.15.2.jar//ch.njol.skript.SkriptEventHandler$PriorityListener.lambda$new$0(SkriptEventHandler.java:36)
[07:37:48 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[07:37:48 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:71)
[07:37:48 ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
[07:37:48 ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131)
[07:37:48 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
[07:37:48 ERROR]: #!#!     at org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:1006)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerPlayer.die(ServerPlayer.java:1180)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.entity.LivingEntity.hurtServer(LivingEntity.java:1545)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.entity.player.Player.hurtServer(Player.java:741)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerPlayer.hurtServer(ServerPlayer.java:1321)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.ServerExplosion.hurtEntities(ServerExplosion.java:515)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.ServerExplosion.explode(ServerExplosion.java:636)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerLevel.explode0(ServerLevel.java:2002)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerLevel.explode0(ServerLevel.java:1970)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerLevel.explode(ServerLevel.java:1952)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.Level.explode(Level.java:1613)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.block.RespawnAnchorBlock.explode(RespawnAnchorBlock.java:183)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.block.RespawnAnchorBlock.useWithoutItem(RespawnAnchorBlock.java:109)
[07:37:48 ERROR]: #!#!     at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.useWithoutItem(BlockBehaviour.java:929)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:560)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:2154)
[07:37:48 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:45)
[07:37:48 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:10)
[07:37:48 ERROR]: #!#!     at net.minecraft.network.PacketProcessor$ListenerAndPacket.handle(PacketProcessor.java:98)
[07:37:48 ERROR]: #!#!     at net.minecraft.network.PacketProcessor.executeSinglePacket(PacketProcessor.java:33)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1543)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.recordTaskExecutionTimeWhileWaiting(MinecraftServer.java:1273)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1389)
[07:37:48 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:304)
[07:37:48 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1474)
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! Skript: 2.15.2 (latest)
[07:37:48 ERROR]: #!#!     Flavor: ketchup
[07:37:48 ERROR]: #!#!     Date: unknown
[07:37:48 ERROR]: #!#! Bukkit: 26.1.2.local-SNAPSHOT
[07:37:48 ERROR]: #!#! Minecraft: 26.1.2
[07:37:48 ERROR]: #!#! Java: 25.0.2 (OpenJDK 64-Bit Server VM 25.0.2+10-LTS)
[07:37:48 ERROR]: #!#! OS: Mac OS X aarch64 26.5
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! Server platform: Paper
[07:37:48 ERROR]: #!#! 
[07:37:48 ERROR]: #!#! Current node: null
[07:37:48 ERROR]: #!#! Current item: set {_boom} (as java.lang.Object) to {-boom::%{_loc} (as java.lang.Object)%} (as java.lang.Object)
[07:37:48 ERROR]: #!#! Current trigger: death of player (death of player) (test.sk, line 11)
[07:37:48 ERROR]: #!#! Thread: Server thread
[07:37:48 ERROR]: #!#! Language: english
[07:37:48 ERROR]: #!#! Link parse mode: DISABLED
[07:37:48 ERROR]: #!#! End of Error.
[07:37:48 ERROR]: #!#! 

```

### Other

No world check is done:
https://github.com/SkriptLang/Skript/blob/ae3da5cee94b9dd8c9e8a63be2f84f12f9e39770/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java#L134-L136

Whereas it is done in toString:
https://github.com/SkriptLang/Skript/blob/ae3da5cee94b9dd8c9e8a63be2f84f12f9e39770/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java#L128-L130

### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

--- a/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java
+++ b/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java
@@ -126,13 +126,13 @@ public class LocationClassInfo extends ClassInfo<Location> {
 
 		@Override
 		public String toString(Location loc, int flags) {
-			String worldPart = loc.getWorld() == null ? "" : " in '" + loc.getWorld().getName() + "'"; // Safety: getWorld is marked as Nullable by spigot
+			String worldPart = loc.getWorld() == null ? "" : " in '" + loc.getWorld().getName() + "'"; // Safety: getWorld is marked as Nullable by Paper
 			return "x: " + Skript.toString(loc.getX()) + ", y: " + Skript.toString(loc.getY()) + ", z: " + Skript.toString(loc.getZ()) + ", yaw: " + Skript.toString(loc.getYaw()) + ", pitch: " + Skript.toString(loc.getPitch()) + worldPart;
 		}
 
 		@Override
 		public String toVariableNameString(Location loc) {
-			String worldPart = loc.getWorld() == null ? "" : loc.getWorld().getName() + ":"; // Safety: getWorld is marked as Nullable by spigot
+			String worldPart = loc.getWorld() == null ? "" : loc.getWorld().getName() + ":"; // Safety: getWorld is marked as Nullable by Paper
 			return worldPart + loc.getX() + "," + loc.getY() + "," + loc.getZ();
 		}
 

--- a/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java
+++ b/src/main/java/org/skriptlang/skript/bukkit/types/LocationClassInfo.java
@@ -132,7 +132,8 @@ public class LocationClassInfo extends ClassInfo<Location> {
 
 		@Override
 		public String toVariableNameString(Location loc) {
-			return loc.getWorld().getName() + ":" + loc.getX() + "," + loc.getY() + "," + loc.getZ();
+			String worldPart = loc.getWorld() == null ? "" : loc.getWorld().getName() + ":"; // Safety: getWorld is marked as Nullable by spigot
+			return worldPart + loc.getX() + "," + loc.getY() + "," + loc.getZ();
 		}
 
 		@Override

--- a/src/test/skript/tests/bukkit/types/Location.sk
+++ b/src/test/skript/tests/bukkit/types/Location.sk
@@ -1,0 +1,24 @@
+test "Location ClassInfo":
+	set {_loc} to location(1,1,1, world("world"))
+	assert {_loc} is set with "Location variable should be set"
+
+	set {_string} to "%{_loc}%"
+	assert {_string} is set with "Location should be converted to string"
+	assert {_string} = "x: 1, y: 1, z: 1, yaw: 0, pitch: 0 in 'world'" with "Location should properly be converted to string"
+
+	set {_var::%{_loc}%} to 1
+	assert {_var::%{_loc}%} is set with "Location should be able to be used in a variable as an index"
+
+	delete {_loc}
+	delete {_string}
+
+	set {_loc} to location(1,1,1, {_null_world})
+	assert {_loc} is set with "Location variable should be set"
+
+	set {_string} to "%{_loc}%"
+	assert {_string} is set with "Location should be converted to string"
+	assert {_string} = "x: 1, y: 1, z: 1, yaw: 0, pitch: 0" with "Location should properly be converted to string"
+
+	set {_var::%{_loc}%} to 1
+	# Reference issue 8661
+	assert {_var::%{_loc}%} is set with "Location without a world should be able to be used in a variable as an index"

